server/plugin.go

```diff
@@ -265,3 +266,28 @@ func (p *Plugin) handleAutoCallsRecording(post *model.Post, postingUser *model.U
 
 	return nil
 }
+
+func (p *Plugin) FileWillBeUploaded(c *plugin.Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string) {
+	if info.Content != "" {
+		return nil, ""
+	}
+
+	if !p.getConfiguration().Config.EnableAudioContentExtraction {
+		return nil, ""
+	}
+
+	if !strings.HasPrefix(info.MimeType, "audio/") && !strings.HasPrefix(info.MimeType, "video/") {
+		return nil, ""
+	}
+
+	if err := p.checkUsageRestrictionsForUser(info.CreatorId); err != nil {
```

Comment:
Can we check channel restrictions here too? Or is it not assigned a channel yet when we get it here?

Reply:
Yes, exactly, You upload the file without the channel assigned yet, we start doing all the processing, and then we assign the post later, so we don't have enough information for the channel permission checking.

Reply:
Hmm. Unfortunate. We won't be able to enable this on community. I don't think we should add code that unexpectedly sends the contents of files in private channels or on teams where you have the plugin disabled. Can we disable this feature when channel/team restrictions are enabled? Also want to add a note in the system console about that behavior.

Reply:
Either one of those could be good. I can see a "FileUpdated" hook being generally useful.